server.go

```diff
@@ -24,9 +24,10 @@ type Server struct {
 	listener      net.Listener
 	wg            waitGroup
 	quit          chan interface{}
-	isStarted     bool
-	PortNumber    int
+	started       bool
+	portNumber    int
 	quitTimeout   chan interface{}
+	sync.Mutex
```

Comment:
Could we use `sync.RWMutex` instead of `sync.Mutex`? `sync.RWMutex` does not block reads unless there is a call to write, therefore, making it a more favored choice to use when accessing a resource. What do you think?

Reply:
I think that it's best to only use `sync.RWMutex` when it's proven that there is a actually need for it. Using a regular `sync.Mutex` will always work, and the way of using it is simple and foolproof. There is no need for a premature optimisation, that's why I chose this.

Reply:
As we have used `sync.RWMutex` in one place, probaly good to use it in another place?

Reply:
Actually, we should change the `RWMutex` → `Mutex` on line 128 because you aren't making use or `Rlock()`/`RUnlock()`

Reply:
Yup, I'll update it. Thanks)